src/CMakeLists.txt

```diff
@@ -47,14 +47,20 @@ set(UMF_SOURCES_MACOSX
     libumf_linux.c
 )
 
+# Compile definitions for UMF library
+set(UMF_COMPILE_DEFINITIONS "")
+
 if(UMF_BUILD_OS_MEMORY_PROVIDER)
     set(UMF_SOURCES_LINUX ${UMF_SOURCES_LINUX}
         provider/provider_os_memory.c
         provider/provider_os_memory_linux.c
         memory_targets/memory_target_numa.c
-        memspaces/memspace_numa.c)
+        memspaces/memspace_numa.c
+        memspaces/memspace_host.c)
     if(LINUX)
         set(UMF_LIBS ${UMF_LIBS} hwloc)
+        # Currently, OS provider is only built for Linux
+        set(UMF_COMPILE_DEFINITIONS ${UMF_COMPILE_DEFINITIONS} "UMF_BUILD_OS_MEMORY_PROVIDER")
```

Comment:
in "benchmark\CMakeLists.txt" there is a similar definition:

if (UMF_BUILD_OS_MEMORY_PROVIDER)
	target_compile_definitions(ubench PRIVATE UMF_BUILD_OS_MEMORY_PROVIDER=1)
endif()

could you add this definition the same way? In the future I think we should make this defines PUBLIC and cleanup cmake files but in the separate PR. You could even add a TODO for that

Reply:
Done.